ts/main.ts

```diff
@@ -1,65 +1,171 @@
+const API_KEY = 'd96ee95088b9438aa8c07a785c81b6e9';
+
 interface Team {
-  TeamID: number;
+  TeamId: number;
   Name: string;
   WikipediaLogoUrl: string;
 }
 
+const TEAM_NAMES = [
+  'Manchester United FC',
+  'FC Barcelona',
+  'Real Madrid CF',
+  'Arsenal FC',
+  'Paris Saint-Germain FC',
+  'Liverpool FC',
+];
+
+interface Player {
+  PlayerID: number;
+  CommonName: string;
+  Position: string;
+  TeamID: number;
+  Jersey: number;
+}
+
+const API_URLS = {
+  EPL: 'https://api.sportsdata.io/v4/soccer/scores/json/Teams/EPL',
+  ESP: 'https://api.sportsdata.io/v4/soccer/scores/json/Teams/ESP',
+  FRL1: 'https://api.sportsdata.io/v4/soccer/scores/json/Teams/FRL1',
+};
+
+const POSITION_MAP: { [key: string]: string } = {
+  D: 'Defender',
+  A: 'Attacker',
+  M: 'Midfielder',
+};
+
+async function fetchTeams(apiUrl: string): Promise<Team[]> {
+  const response = await fetch(`${apiUrl}?key=${API_KEY}`);
+  const data = await response.json();
+  return data.filter((team: Team) => TEAM_NAMES.includes(team.Name)) as Team[];
+}
+
 const teamsContainer: HTMLElement | null =
   document.getElementById('teams-container');
+const starting11Container: HTMLElement | null = document.getElementById(
+  'starting11-container',
+);
+const teamLogoElement: HTMLImageElement | null = document.getElementById(
+  'team-logo',
+) as HTMLImageElement;
+const titleElement: HTMLElement | null = document.getElementById('title');
 
-const eplApiUrl: string =
-  'https://api.sportsdata.io/v4/soccer/scores/json/Teams/EPL?key=d96ee95088b9438aa8c07a785c81b6e9';
-const espApiUrl: string =
-  'https://api.sportsdata.io/v4/soccer/scores/json/Teams/ESP?key=d96ee95088b9438aa8c07a785c81b6e9';
-const frlApiUrl: string =
-  'https://api.sportsdata.io/v4/soccer/scores/json/Teams/FRL1?key=d96ee95088b9438aa8c07a785c81b6e9';
-
-async function fetchTeamLogos(
-  apiUrl: string,
-  filterTeams?: string[],
-): Promise<void> {
+const starting11ApiUrls: { [key: string]: string } = {
+  'Arsenal FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/EPL/509?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Manchester United FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/EPL/517?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Liverpool FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/EPL/515?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'FC Barcelona':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/ESP/559?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Real Madrid CF':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/ESP/605?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Paris Saint-Germain FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/FRL1/600?key=d96ee95088b9438aa8c07a785c81b6e9',
+};
```

Comment:
Is there any way you can do this without having all of your endpoints hard coded? This makes your project very closed off in only being what you decide which may not be what a user wants.

I see that it does change the teams or leagues with the EPL, ESP, or FRL!, but can you get the number after some other way? Does it come from your first API call and you can attach it to an element and access it that way?